plugins/scaffolder-backend/src/scaffolder/stages/publish/helpers.ts

```diff
@@ -67,3 +68,63 @@ export async function initRepoAndPush({
     remote: 'origin',
   });
 }
+
+type BranchProtectionOptions = {
+  client: Octokit;
+  owner: string;
+  repoName: string;
+  isRetry?: boolean;
+};
+
+export const enableBranchProtectionOnDefaultRepoBranch = async ({
+  repoName,
+  client,
+  owner,
+  isRetry = false,
```

Comment:
A lil bit awkward to have this here for the recursion, could use a loop instead. It's internal stuff though so no biggie

Reply:
Thanks! I'd appreciate it if we could maybe add a little information about _why_ it's awkward and why using a loop instead would be more beneficial. By choosing to do it this way, these are facts that are not immediately obvious to me so it'd be helpful to read more. 😄

Reply:
I'm assuming that pointing the awkwardness out is often enough to understand it, but if it isn't a followup question is no problem.

It's awkward because the external signature of the function has a parameter that should never be set and potentially also confuses the caller. It somewhat implies that the caller should call this function again on failure, with the parameter set to true. Since it's an internal helper there's not that much harm done by it though, but it still has potential to cause some bugs in the future.